### PR TITLE
[Merged by Bors] - feat: semi-uniqueness of limits in seminormed spaces

### PR DESCRIPTION
**Pull request by eric-wieser**
[Zulip thread](https://leanprover.zulipchat.com/#narrow/stream/217875-Is-there-code-for-X.3F/topic/Tendsto.20in.20seminormed.20spaces/near/472351143)

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/MetricSpace/Pseudo/Basic.lean [hunk 1]
```diff
@@ -172,6 +172,14 @@ open Metric
...
   rw [EMetric.inseparable_iff, edist_nndist, dist_nndist, ENNReal.coe_eq_zero, NNReal.coe_eq_zero]
 
+alias ⟨Inseparable.dist_eq_zero, _⟩ := Metric.inseparable_iff
+
+/-- A weaker version of `tendsto_nhds_unique` for `PseudoMetricSpace`. -/
+theorem tendsto_nhds_unique_dist {X Y : Type*} [PseudoMetricSpace X]
+    {f : Y → X} {l : Filter Y} {a b : X} [NeBot l]
+    (ha : Tendsto f l (𝓝 a)) (hb : Tendsto f l (𝓝 b)) : dist a b = 0 :=
+  (tendsto_nhds_unique_inseparable ha hb).dist_eq_zero
+
```
Comment (eric-wieser) [inline Mathlib/Topology/MetricSpace/Pseudo/Basic.lean:182]: I could be persuaded to drop this one if you think it's not useful.

File: Mathlib/Topology/MetricSpace/Pseudo/Basic.lean [hunk 2]
```diff
@@ -169,8 +169,21 @@ end Metric
...
+alias ⟨Inseparable.nndist_eq_zero, _⟩ := Metric.inseparable_iff_nndist
+
 theorem Metric.inseparable_iff {x y : α} : Inseparable x y ↔ dist x y = 0 := by
-  rw [EMetric.inseparable_iff, edist_nndist, dist_nndist, ENNReal.coe_eq_zero, NNReal.coe_eq_zero]
+  rw [Metric.inseparable_iff_nndist, dist_nndist, NNReal.coe_eq_zero]
+
+alias ⟨Inseparable.dist_eq_zero, _⟩ := Metric.inseparable_iff
+
+/-- A weaker version of `tendsto_nhds_unique` for `PseudoMetricSpace`. -/
+theorem tendsto_nhds_unique_dist {X Y : Type*} [PseudoMetricSpace X]
```
Comment (urkud) [inline Mathlib/Topology/MetricSpace/Pseudo/Basic.lean:183]: You already have a `PseudoMetricSpace` in the context. Please reuse the variable.

File: Mathlib/Topology/Separation.lean [hunk 3]
```diff
@@ -1045,6 +1045,16 @@ theorem r1Space_iff_inseparable_or_disjoint_nhds {X : Type*} [TopologicalSpace X
   ⟨fun _h x y ↦ (specializes_or_disjoint_nhds x y).imp_left Specializes.inseparable, fun h ↦
     ⟨fun x y ↦ (h x y).imp_left Inseparable.specializes⟩⟩
 
+theorem inseparable_of_nhds_neBot {x y : X} (h : NeBot (𝓝 x ⊓ 𝓝 y)) :
```
Comment (urkud) [inline Mathlib/Topology/Separation.lean:1048]:
```suggestion
theorem Inseparable.of_nhds_neBot {x y : X} (h : NeBot (𝓝 x ⊓ 𝓝 y)) :
```

File: Mathlib/Topology/Separation.lean [hunk 4]
```diff
@@ -1045,6 +1045,16 @@ theorem r1Space_iff_inseparable_or_disjoint_nhds {X : Type*} [TopologicalSpace X
...
+    Inseparable x y :=
+  (r1Space_iff_inseparable_or_disjoint_nhds.mp ‹_› _ _).resolve_right fun h' => h.ne h'.eq_bot
+
+/-- A weaker version of `tendsto_nhds_unique` for `R1Space`. -/
+theorem tendsto_nhds_unique_inseparable {X Y : Type*} [TopologicalSpace X] [R1Space X]
```
Comment (urkud) [inline Mathlib/Topology/Separation.lean:1053]: Please reuse the available type variable(s). Also, do we use latin letters for spaces without structure (`Y` in this case)?

Comment (eric-wieser) [reply Mathlib/Topology/Separation.lean]: > Also, do we use latin letters for spaces without structure (`Y` in this case)?

Yes, the rest of the file still does this.

File: Mathlib/Topology/MetricSpace/Pseudo/Basic.lean (hunk 2 above)
Comment (eric-wieser) [reply Mathlib/Topology/MetricSpace/Pseudo/Basic.lean]: Done

**Comment by urkud** [comment]
Thanks! Once CI is happy, please merge this PR using `bors r+`
bors d+

**Comment by eric-wieser** [comment]
bors merge

**Pull request closed without merging**